google-cloud-storage/lib/google/cloud/storage/bucket.rb

```diff
@@ -665,11 +665,23 @@ def create_file file, path = nil, acl: nil, cache_control: nil,
         #                                  issuer: "service-account@gcloud.com",
         #                                  signing_key: key
         #
+        # @example Using the headers option:
+        #   require "google/cloud/storage"
+        #
+        #   storage = Google::Cloud.storage
+        #
+        #   bucket = storage.bucket "my-todo-app"
+        #   shared_url = bucket.signed_url "avatars/heidi/400x400.png",
+        #                                  headers: {
+        #                                    "x-goog-acl" => "private",
+        #                                    "x-goog-meta-foo" => "bar,baz"
```

Comment:
The headers will be downcased, per the [spec](https://cloud.google.com/storage/docs/access-control/signed-urls#about-canonical-extension-headers). That's why the various tests are using different casing, to ensure that it works correctly.

I don't know how the headers should be presented in the documentation, however. What do you think?